### PR TITLE
Admin Create Posting Shifts Page UI

### LINKED ISSUE #58: Admin create posting "Add Shifts" form and page
## User Story and Context

As an admin (Sistering employee), I need to add the specific timeslots for shifts so volunteers can identify if they are available for the opportunity.

### Product
Please see #42. Note that the attached design in #42 is outdated, here is a more recent version, however you should confirm with @amandatdu before proceeding.

**This ticket covers the parts enclosed in the red rectangle only**:

**Figma link:** https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Admin-Experience?node-id=186%3A2199

Please confirm with @amandatdu on which designs to use.

### Technical
* Create both a form component (in `CreatePostingShifts.tsx` under `frontend/src/components/admin/posting`) and mount to the corresponding page (located in `frontend/src/components/pages/admin/posting/CreatePostingShiftsPage.tsx`)
* The page should have the route `/admin/posting/create/shifts`
* Form data should be written to global state using the context API so that it can easily be shared with the "Review Posting" page. **It does not need to be sent to the backend yet.** Trace through the "Edit Team" example from starter code to understand this state management pattern, here is the [starting point](https://github.com/uwblueprint/sistering/blob/main/frontend/src/App.tsx#L38-L40)
* **Input validation occurs when user presses "Next"**

## Dev Notes

**Suggested components to use:**

**Chakra docs:**
* [Select](https://chakra-ui.com/docs/form/select)
* [Input](https://chakra-ui.com/docs/form/input)

For input validation, you will need to pair the components with [FormControl](https://chakra-ui.com/docs/form/form-control).

### PR DESCRIPTION
**Pull request by jiarucao**
## Ticket link

Closes #58

## Implementation description
* Create form component to allow admins to create posting shifts 

## Steps to test
1. Run the application locally or use the preview link below
2. Log in and click on the "Create Shifts" tab on the default page 
3. Fill in the fields and test that the field errors work as expected (no empty fields, and start date should not be later than end date)
4. Verify that the data is being reflected in the global context by going back to the default page (using the "Go Back" button) and checking the start date, end date, and recurrence intervals match what you inputed

## What should reviewers focus on?
* Does the context work correctly?
* Do the field validations work as expected?

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/admin/posting/CreatePostingShifts.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            </Text>
+          </Circle>
+          <Text textStyle="heading">Scheduling Time Slots</Text>
+        </HStack>
+        <VStack spacing={30} alignItems="flex-start" px={2}>
+          <Text textStyle="caption">
+            {ADMIN_POSTING_CREATE_SCHEDULING_TIME_SLOTS}
+          </Text>
+          <FormControl isRequired isInvalid={recurrenceIntervalError}>
+            <FormLabel textStyle="heading">Reoccurance Frequency</FormLabel>
```
Comment (sherryhli) [inline frontend/src/components/admin/posting/CreatePostingShifts.tsx:102]: Pretty sure this was an unintentional typo in the Figma, let's change this to "Recurrence Frequency"

File: frontend/src/components/admin/posting/CreatePostingShifts.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+            <Flex alignItems="flex-start">
+              <VStack spacing={2} alignItems="flex-end">
+                <Select
+                  placeholder="How often will this occur?"
+                  size="sm"
+                  width="425px"
+                  isRequired
```
[the remark this answers is not in the document]
Comment (jiarucao) [reply frontend/src/components/admin/posting/CreatePostingShifts.tsx:109]: nope, will remove!

File: frontend/src/components/admin/posting/CreatePostingShifts.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+              Select Start and End Dates
+            </Text>
+            <Flex>
+              <HStack spacing={5} alignItems="flex-start">
+                <FormControl isRequired isInvalid={startDateError}>
+                  <Flex>
+                    <FormLabel textStyle="heading" fontSize="18px">
+                      From
+                    </FormLabel>
+                    <VStack spacing={2} alignItems="flex-end">
```
[the remark this answers is not in the document]
Comment (jiarucao) [reply frontend/src/components/admin/posting/CreatePostingShifts.tsx:135]: This `VStack` keeps the error message right aligned under the input field :)

File: frontend/src/components/admin/posting/CreatePostingShifts.tsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+                      </FormErrorMessage>
+                    </VStack>
+                  </Flex>
+                </FormControl>
+                <FormControl isRequired isInvalid={endDateError}>
+                  <Flex>
+                    <FormLabel textStyle="heading" fontSize="18px">
+                      To
+                    </FormLabel>
+                    <VStack spacing={2} alignItems="flex-end">
```
[the remark this answers is not in the document]
Comment (jiarucao) [reply frontend/src/components/admin/posting/CreatePostingShifts.tsx:156]: error message alignment again^

File: frontend/src/components/admin/posting/CreatePostingShifts.tsx [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+          <Text textStyle="heading">Scheduling Time Slots</Text>
+        </HStack>
+        <VStack spacing={30} alignItems="flex-start" px={2}>
+          <Text textStyle="caption">
+            {ADMIN_POSTING_CREATE_SCHEDULING_TIME_SLOTS}
+          </Text>
+          <FormControl isRequired isInvalid={recurrenceIntervalError}>
+            <FormLabel textStyle="heading">Reoccurance Frequency</FormLabel>
+            <Flex alignItems="flex-start">
+              <VStack spacing={2} alignItems="flex-end">
```
[the remark this answers is not in the document]
Comment (jiarucao) [reply frontend/src/components/admin/posting/CreatePostingShifts.tsx:104]: This is used to align the error message for the selector

File: frontend/src/components/admin/posting/CreatePostingShifts.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sherryhli) [reply frontend/src/components/admin/posting/CreatePostingShifts.tsx:135]: Got it, thanks for explaining!

**Review by sherryhli — APPROVED** [review]
Beautiful, LGTM! 🚢

**Pull request merged**